apps/src/dance/lab2/ProgramExecutor.ts

```diff
@@ -115,16 +114,26 @@ export default class ProgramExecutor {
       return;
     }
 
-    this.hooks.runUserSetup();
+    const previewDraw = () => {
+      this.nativeAPI.setEffectsInPreviewMode(true);
+
+      // the user setup hook initializes effects,
+      // needs to happen in preview mode for some effects (eg, tacos)
```

Comment:
And just checking my understanding - doesn't this need to happen for all effects?

Reply:
Only some effects have preview-only customizations in their `init` functions (tacos is one example). Some of them work fine even if we didn't move the init step here

Reply:
Gotcha! I was thinking of the 'the user setup hook initializes effects' part of the comment for my question. Maybe break comment into two comments and the 'needs to happen in preview mode' could be above where preview mode is set to `true`?

Reply:
Maybe I'll just remove this comment, I think it's clear from the code that the whole block needs to happen in preview mode. Calling out this line specifically probably just muddies things

Reply:
Sounds good - thanks!